dashboard/app/helpers/aichat_safety_helper.rb

```diff
@@ -34,18 +36,28 @@ def find_toxicity(role, text, locale)
 
     private def openai_safety_check(text)
       details = nil
+      start_time = Time.now
+      report_openai_safety_check("#{self.class.name}.OpenaiStart")
       # Try twice in case of network errors or model not correctly following directions and
       # replying with something other valid expected output.
+      attempts = 1
       Retryable.retryable(tries: 2) do
         openai_response = OpenaiChatHelper.request_safety_check(text, get_safety_system_prompt)
         evaluation = JSON.parse(openai_response)['choices'][0]['message']['content']
-        raise "Unexpected response from OpenAI: #{evaluation}" unless VALID_EVALUATION_RESPONSES_SIMPLE.include?(evaluation)
+        unless VALID_EVALUATION_RESPONSES_SIMPLE.include?(evaluation)
+          report_openai_safety_check("#{self.class.name}.OpenaiInvalidResponse")
+          attempts += 1
+          raise "Unexpected response from OpenAI: #{evaluation}"
+        end
         if evaluation == 'INAPPROPRIATE'
           details = {
             evaluation: evaluation
           }
         end
       end
+      report_openai_safety_check("#{self.class.name}.OpenaiFinish")
```

Comment:
Wondering if we should also report num_attempts with the finish metric...it might be good to see a ratio of how many successful requests took 2 attempts vs 1?

Reply:
Sure - will update